### PR TITLE
[WIP] Add the `cell_borders()` helper and update implementation of the stylizing helper fcns

### PR DESCRIPTION
**Pull request by rich-iannone**
This adds the `cell_borders()` helper function for selectively modifying cell borders in a `tab_style()` call. The existing `cells_styles()` helper was reworked as `cell_text()` and the single argument dealing with the cell background color was put in `cell_fill()`.

These three stylizing helper functions can be placed in a `list()` of calls for the `style` argument of `tab_style()`, allowing us to target cells and perform multiple stylizing operations on the targets at once.

Here is an example:

```r
library(gt)

exibble %>%
  gt() %>%
  tab_style(
    style = list(
      cell_borders(
        selection = c("left", "right"),
        color = "green",
        style = "solid",
        weight = px(3)
      ),
      cell_fill(color = "lightgray"),
      cell_text(color = "steelblue", font = "HelveticaNeue")
    ),
    locations = cells_data(columns = vars(row, char))
  )
```

exibble_styles

Fixes https://github.com/rstudio/gt/issues/258.
Fixes https://github.com/rstudio/gt/issues/245.

cc: https://github.com/ddsjoberg/gtsummary/issues/78

### CODE REVIEW

File: R/data_color.R [hunk 1]
```diff
@@ -129,7 +129,7 @@ data_color <- function(data,
                        columns,
                        colors,
                        alpha = NULL,
-                       apply_to = "bkgd",
+                       apply_to = "fill",
```
Comment (jcheng5) [inline R/data_color.R]: Note: BREAKING change. At the very least, consider putting a prominent note in NEWS. Consider checking for `"bkgd"` in the body of the function, and respond appropriately (maybe warn, then act as if they asked for `"fill"`?).

**Comment by jcheng5** [comment]
This is an incremental change on top of a whole pile of code I have not reviewed yet. Besides normal code quality stuff, I'd want to focus on whether you have a vision for making the styling features portable between HTML, LaTeX, and RTF.

File: R/data_color.R (hunk 1 above)
Comment (jcheng5) [reply R/data_color.R]: Same with the move from `cells_styles` to `cell_text` and `cell_fill`, do you have a plan for how to notify?

**Comment by rich-iannone** [comment]
Using #239 instead.

**Pull request closed without merging**